lib/internal/url.js

```diff
@@ -1398,14 +1398,14 @@ function getPathFromURLPosix(url) {
   return decodeURIComponent(pathname);
 }
 
-function fileURLToPath(path) {
+function fileURLToPath(path, parentURL) {
```

Comment:
I think base is a better name for parentURL (aligns with URL constructor parameters)

Reply:
I feel hesitant about adding a new parameter to this function just for this error use case. I feel like the API doesn’t make sense for the general app `import { fileURLToPath } from 'node:'url'` common use case.

We could make a separate function `fileURLToPathWithParent` or something, that wraps this one; or can’t there be some other way to pass along the parent URL without it needing to go through `node:url` methods?

Reply:
> I feel hesitant about adding a new parameter to this function just for this error use case.

+1. i don't like it either.

> We could make a separate function fileURLToPathWithParent or something, that wraps this one

how? it would still need to pass down the arg.

>can’t there be some other way to pass along the parent URL without it needing to go through node:url methods?

i'm not aware of any of our errors doing that. if we can, that sounds like an avenue worth exploring :)